enhancements/microshift/rhoai-model-serving-on-microshift.md

```diff
@@ -0,0 +1,339 @@
+- [RHOAI's overrides](https://github.com/red-hat-data-services/kserve/blob/master/config/overlays/odh/inferenceservice-config-patch.yaml)
+
+While we can recommend users to create a manifest that will override the
+ConfigMap to their liking, there's one setting that we could handle better:
+`ingress.ingressDomain`.  In the example it has value of `example.com` and it
+might be poor UX to require every customer to create a new ConfigMap just to change this.
+
+Possible solution is to change our manifest handling logic, so that MicroShift 
+uses kustomize's Go package to first render the manifest, then template it,
+and finally apply. For this particular `ingress.ingressDomain` we could reuse value of
+`dns.baseDomain` from MicroShift's config.yaml.
+
+
+### How to deliver ServingRuntime CRs
+
+From [kserve documentation](https://kserve.github.io/website/master/modelserving/servingruntimes/):
+
+> KServe makes use of two CRDs for defining model serving environments:
+> 
+> ServingRuntimes and ClusterServingRuntimes
+> 
+> The only difference between the two is that one is namespace-scoped and the other is cluster-scoped.
+> 
+> A ServingRuntime defines the templates for Pods that can serve one or more 
+> particular model formats. Each ServingRuntime defines key information such as
+> the container image of the runtime and a list of the model formats that the
+> runtime supports. Other configuration settings for the runtime can be conveyed
+> through environment variables in the container specification.
+
+RHOAI approach to ServingRuntimes:
+- ClusterServingRuntimes are not supported (the CRD is not created).
+- Each usable ServingRuntime is wrapped in Template and resides in RHOAI's namespace.
+- When user uses RHOAI Dashboard to serve a model, they must select a runtime
+  from a list (which is constructed from Templates holding ServingRuntime)
+  and provide information about the model.
+- When the user sends the form, Dashboard creates a ServingRuntime from the
+  Template in the user's Data Science Project (effectively Kubernetes namespace),
+  and assembles the InferenceService CR.
+
+Problem: how to provide user with ServingRuntimes without creating unnecessary obstacles.
+
+In any of the following solutions, we need to drop the `Template` container to
+get only the `ServingRuntime` CR part.
+
+Potential solutions so far:
+- Include ServingRuntimes in a RPM as a kustomization manifest (might be 
+  `microshift-kserve`, `microshift-rhoai-runtimes`, or something else) using a
+  specific predefined namespace.
+  - This will force users to either use that namespace for serving, or they can
+    copy the SRs to their namespace (either at runtime, or by including it in
+    their manifests).
+- Change ServingRuntimes to ClusterServingRuntimes and include them in an RPM,
+  so they're accessible from any namespace (MicroShift is intended for
+  single-user operation anyway).
```

Comment:
From a MicroShift user perspective this is probably the best option and we are considering to adopt ClusterServingRuntime in the future but not sure when

Reply:
Yup, I also think it'd be the most convenient